uportal-war/src/main/webapp/WEB-INF/flows/edit-portlet/portletConfig.jsp

```diff
@@ -0,0 +1,891 @@
+                NameTitleModel,
+                nameTitleModel,
+                nameTitleMismatchTimer,
+                nameChangedFn,
+                titleChangedFn,
+                fnameChangedFn;
+
+        /**
+         * Define a backbone model to capture the name/title/fname values and
+         * orchestrate their updates.  Changes on the page should update the model
+         * and allow listeners attached to the model to handle propagating the changes
+         * to the appropriate places in the UI
+         */
+        NameTitleModel = Backbone.Model.extend({
+            defaults: {
+                name: '',
+                title: '',
+                titleWarn: false,
+                titleEdited: false,
+                fname: '',
+                fnameEdited: false,
+                timeout: '5000'
+            },
+
+
+            /**
+             * Initialize the model.
+             */
+            initialize: function () {
+                this.on('change:title', this._updateTitle, this);
+                this.on('change:name', this._updateName, this);
+                this.on('change:fname', this._updateFName, this);
+            },
+
+
+            /**
+             * import existing settings into the model.  Will attempt to determine if the
+             * title and name match and if the title and fname are in sync.  If so, it will
+             * enable auto-updates to those fields.  If not, changing title will not affect
+             * the fields that appear to have been modified.
+             */
+            import: function (name, title, fname) {
+                var calcFName, titleEdited, fnameEdited;
+
+                if (title && name) {
+                    if (title !== name) {
+                        titleEdited = true;
+                    }
+                }
+
+                if (title && fname) {
+                    calcFName = this._toFName(title);
+                    if (fname !== calcFName) {
+                        fnameEdited = true;
+                    }
+                }
+
+                this.set({
+                    name: name,
+                    title: title,
+                    fname: fname,
+                    titleEdited: titleEdited,
+                    fnameEdited: fnameEdited
+                }, {validate: true});
+            },
+
+
+            /**
+             * Do validation to check for name/title mismatches and set the 'titleWarn' attribute.
+             */
+            validate: function (attrs, options) {
+                var finalAttrs = _.extend({}, this.attributes, attrs);
+                this.set('titleWarn', finalAttrs.title !== finalAttrs.name);
+            },
+
+
+            /**
+             * Given a title, calculate the fname to use.
+             */
+            _toFName: function (name) {
+                var tmp = name.toLowerCase();
+                return tmp ? tmp.replace(/[^a-z0-9]+/g, '-') : '';
```

Comment:
I think this will be fine for international. They might have a few extra dashes, but no harm.